src/Avalonia.Base/Data/BindingValue.cs

```diff
@@ -190,7 +190,7 @@ public BindingValue<T> WithValue(T value)
         /// Gets the value of the binding value if present, otherwise the default value.
         /// </summary>
         /// <returns>The value.</returns>
-        public T GetValueOrDefault() => HasValue ? _value : default;
+        public T GetValueOrDefault() => HasValue ? _value : default!;
```

Comment:
I think these warnings would be better fixed with the `[return: MaybeNull]` attribute.